src/services/popover/popover_positioning.js

```diff
@@ -77,12 +77,22 @@ export function findPopoverPosition({ anchor, popover, position, buffer=16, offs
    * if position = "right" the order is right, left, top, bottom
    */
 
-  let iterationPosition = position;
-
-  let bestFit = -Infinity;
-  let bestPosition = null;
+  const iterationPositions = [
+    position,                                           // Try the user-desired position first.
+    positionComplements[position],                      // Try the complementary position.
+    positionSubstitutes[position],                      // Switch to the cross axis.
+    positionComplements[positionSubstitutes[position]], // Try the complementary position on the cross axis.
+  ];
+
+  const {
+    bestPosition,
+  } = iterationPositions.reduce(({ bestFit, bestPosition }, iterationPosition) => {
```

Comment:
I like the encapsulation of using `reduce` because you don't need to mutate external variables, but it's a little inefficient because it always loops 4 times, even if the best fit has already been found. I don't mind if you want to switch this back to a `for` loop.